src/zimscraperlib/zim/creator.py

```diff
@@ -146,7 +150,30 @@ def config_indexing(
         self.__indexing_configured = True
         return self
 
+    def _get_illustration_metadata(
+        self,
+        name: str,
+        value: bytes | str | datetime.datetime | datetime.date | Iterable[str],
+    ) -> str | None:
+        """Return image format for debug logging of illustration metadata"""
+        if not name.startswith("Illustration_"):
+            return None
+        if not isinstance(value, bytes):
+            return f"Unexpected image datatype, {value.__class__.__name__}"
+        return (
+            format_for(io.BytesIO(value), from_suffix=False)
+            or f"Unknown image format, {len(value)} bytes"
+        )
+
+    def _log_metadata(self):
+        if logger.isEnabledFor(DEBUG):
```

Comment:
Not sure it's really worth skipping those tiny computations when not in debug mode but should we want to keep it, we should condition the `log_metadata()` call instead of having this condition inside.